org/mozilla/jss/ssl/javax/JSSParameters.java

```diff
@@ -54,6 +56,11 @@ public JSSParameters(SSLParameters downcast) {
         if (downcast.getNeedClientAuth()) {
             setNeedClientAuth(downcast.getNeedClientAuth());
         }
+
+        String[] alpn = JDKCompat.SSLParametersHelper.getApplicationProtocols(downcast);
```

Comment:
Is this necessary? The `SSLParameters.getApplicationProtocols()` doesn't seem to change in different Java versions:
* https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLParameters.html#getApplicationProtocols--
* https://docs.oracle.com/javase/9/docs/api/javax/net/ssl/SSLParameters.html#getApplicationProtocols--
* https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/SSLParameters.html#getApplicationProtocols()

Reply:
See above; yes, because support in JDK8 is really, really new. :)